src/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 add_subdirectory(doorman)
 add_subdirectory(chat)
-add_executable(xania
+add_subdirectory(test)
+add_library(xania_lib STATIC
```

Comment:
Aha, a static library at last :)

Reply:
Megapedantry but I'm not bothered:  libxania_lib.a == double lib :) It could perhaps just be libxania.a

Reply:
Sadly we already have a `xania` target in CMake-land :( (the executable). So we either have a "xania" binary or a "xania" lib, but not both

Reply:
You can modify the binary name to fix this. See e.g. 

https://github.com/master-keying/minisat/blob/master/CMakeLists.txt#L52-L57

Reply:
Oh thanks @horenmar: TIL :D